Main.java

```diff
@@ -3,74 +3,145 @@
+            Database.wordSearchFailure(textFilePath, searchKeyword, ERROR_MESSAGE);
             return false;
         } else if (!(file.getName().endsWith(FILE_TYPE_TXT) || file.getName().endsWith(FILE_TYPE_JSON))) {
-            System.out.println("File format not supported");
+            ERROR_MESSAGE = "File format not supported";
+            System.out.println(ERROR_MESSAGE);
+            Database.wordSearchFailure(textFilePath, searchKeyword, ERROR_MESSAGE);
             return false;
         }
         return true;
     }
 
     private static void wordCount(String textFilePath, String searchKeyword) {
 
-        StringBuilder stringBuilder = new StringBuilder();
-        int wordCount = 0;
+        int count = 0;
         String fileData;
         try {
             BufferedReader bufferedReader = Files.newBufferedReader(Paths.get(textFilePath));
-            fileData = bufferedReader.readLine();
-            if (fileData == null) {
-                System.out.println("File does not contain any data");
-                return;
-            } else {
-                while (fileData != null) {
-                    stringBuilder.append(fileData).append("\n");
-                    fileData = fileData.replaceAll(REMOVE_SPECIAL_CHAR, SINGLE_SPACE);
-                    StringTokenizer splitIntoToken = new StringTokenizer(fileData);
-                    while (splitIntoToken.hasMoreTokens()) {
-                        if (searchKeyword.equalsIgnoreCase(splitIntoToken.nextToken())) {
-                            wordCount++;
-                        }
+            while ((fileData = bufferedReader.readLine()) != null) {
+                fileData = fileData.replaceAll(REMOVE_SPECIAL_CHAR, SINGLE_SPACE);
+                StringTokenizer splitIntoToken = new StringTokenizer(fileData);
+                while (splitIntoToken.hasMoreTokens()) {
+                    if (searchKeyword.equalsIgnoreCase(splitIntoToken.nextToken())) {
+                        count++;
                     }
                 }
-                if (wordCount == 0) {
-                    System.out.println("Word does not exist");
-                } else {
-                    System.out.println("Word found");
-                    System.out.println(searchKeyword + " occurs " + wordCount + " times in the file");
-                }
+            }
+            if (count == 0) {
+                ERROR_MESSAGE = "Word does not exist";
+                System.out.println(ERROR_MESSAGE);
+                Database.wordSearchFailure(textFilePath, searchKeyword, ERROR_MESSAGE);
+            } else {
+                System.out.println("Word found");
+                System.out.println(searchKeyword + " occurs " + count + " times in the file");
+                Database.wordSearchSuccess(textFilePath, searchKeyword, count);
             }
         } catch (Exception e) {
             e.printStackTrace();
         }
     }
 }
 
+class Database {

```

Comment:
Why is a different class defined in the same file? Each class should be in its own file